03 - Data Science and Custom Skills/Azure Function Skills/SampleSkills/0_CustomSkillTemplate.cs

```diff
@@ -0,0 +1,171 @@
+/****************************************************************************************
+ * 
+ * This template should make it easy for you to create a custom skill as
+ * it takes care of most of the boilerplate code for serialization and  deserialization.
+ * 
+ * STEPS:
+ * 1. Define the input fields for your skill.
+ *    Modify the InputRecordData class.
+ * 
+ * 2. Define the output fields for your skill.
+ *    Modify the OutputRecordData class.
+ *    
+ * 3. Define what action your will will take to enrich/transform the input into the output.
+ *    Modify the DoWorkHere class.
+ *    
+ ****************************************************************************************/
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+
+
+namespace SampleSkills
+{
+
+    public static class CustomSkillTemplate
+    {
+
+        #region Class used to deserialize the request
+        public class InputRecord
+        {
+            public class InputRecordData
+            {
+                public string myInputField;
+            }
+
+            public string recordId { get; set; }
```

Comment:
You should be able to safely capitalize all of these in order to have proper c# semantics.  See how Brian did it for JFK:
https://github.com/Microsoft/AzureSearch_JFK_Files/blob/master/JfkWebApiSkills/JfkWebApiSkills/WebApiSkill/WebAPISkillContract.cs

Reply:
Got it, I did not realize that Newtonsoft would change the casing for me.